protocol/daemons/server/bridge.go

```diff
@@ -23,30 +21,24 @@ func (server *Server) WithBridgeEventManager(
 	return server
 }
 
-// ExpectBridgeDaemon registers the bridge daemon with the server. This is required
-// in order to ensure that the daemon service is called at least once during every
-// maximumAcceptableUpdateDelay duration. It will cause the protocol to panic if the daemon does not
-// respond within maximumAcceptableUpdateDelay duration.
-func (server *Server) ExpectBridgeDaemon(maximumAcceptableUpdateDelay time.Duration) {
-	server.registerDaemon(types.BridgeDaemonServiceName, maximumAcceptableUpdateDelay)
-}
-
 // AddBridgeEvents stores any bridge events recognized by the daemon
 // in a go-routine safe slice.
 func (s *Server) AddBridgeEvents(
 	ctx context.Context,
 	req *api.AddBridgeEventsRequest,
 ) (
-	*api.AddBridgeEventsResponse,
-	error,
+	response *api.AddBridgeEventsResponse,
+	err error,
 ) {
-	// If the daemon is unable to report a response, there is either an error in the registration of
-	// this daemon, or another one. In either case, the protocol should panic.
-	// TODO(CORE-582): Re-enable this check once the bridge daemon is fixed in local / CI environments.
-	//if err := s.reportResponse(types.BridgeDaemonServiceName); err != nil {
-	//	panic(err)
-	//}
-	if err := s.bridgeEventManager.AddBridgeEvents(req.BridgeEvents); err != nil {
+	// Capture valid responses in metrics.
+	defer func() {
+		if err == nil {
```

Comment:
Not 100% how the closure works, but will this value check the value assigned by [line 41](https://github.com/dydxprotocol/v4-chain/pull/783/files#diff-311868d8f30cb540f3ba861a80d88805bde4b5277b071aff85e4c8e5f401bd76R41)?

Generally, I think it's better to add this right above the return statement. If someone forgets the `err = ` and uses `err :=` then this wont work?

Reply:
Yes, the closure would check the `err` assigned in line 41 since that is what is in scope.

rant: `:=` should not allow creating a new symbol with the same name, it makes reading code very difficult.